software/base/src/test/java/org/apache/brooklyn/entity/software/base/lifecycle/ScriptHelperIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+        entity.start(ImmutableList.of(loc));
+        VanillaSoftwareProcessSshDriver driver = (VanillaSoftwareProcessSshDriver) entity.getDriver();
+        String launchContents = Joiner.on('\n')
+                .join("#!/usr/bin/env bash",
+                        "function trap_handler_command {",
+                        "  echo stopping...",
+                        "  exit 13",
+                        "}",
+                        "trap \"trap_handler_command\" SIGTERM",
+                        "while true; do",
+                        "  sleep 1",
+                        "done");
+        driver.copyResource(ImmutableMap.of(), new ReaderInputStream(new StringReader(launchContents), "UTF-8"), "launch.sh", true);
+        driver.executeLaunchCommand("nohup bash launch.sh > /dev/null &");
+        ScriptHelper stopCommandScriptHelper = driver.stopCommandScriptHelper();
+        stopCommandScriptHelper.execute();
+        Assert.assertEquals(StringUtils.countMatches(stopCommandScriptHelper.getResultStdout(), "is still running"), 1, "SIGTERM should be tried one time");
+        Assert.assertFalse(stopCommandScriptHelper.getResultStdout().contains("stopped with SIGKILL"), "SIGKILL should not be sent after SIGTERM fails.");
+    }
+
+
+    @Test(groups = "Integration")
+    public void testStopWithSigtermIsKilledWithSigKill() {
+        StopCommandSoftwareProcess entity = app.createAndManageChild(EntitySpec.create(StopCommandSoftwareProcess.class, StopCommandSoftwareProcessImpl.class));
+        entity.start(ImmutableList.of(loc));
+        VanillaSoftwareProcessSshDriver driver = (VanillaSoftwareProcessSshDriver) entity.getDriver();
+        String launchContents = Joiner.on('\n')
+                .join("#!/usr/bin/env bash",
+                        "function trap_handler_command {",
+                        "  while true; do",
+                        "    echo \"Do nothing.\"",
+                        "    sleep 1",
+                        "  done",
+                        "}",
+                        "trap \"trap_handler_command\" SIGTERM",
+                        "while true; do",
+                        "  sleep 1",
+                        "done");
+        driver.copyResource(ImmutableMap.of(), new ReaderInputStream(new StringReader(launchContents), "UTF-8"), "launch.sh", true);
+        driver.executeLaunchCommand("nohup bash launch.sh > /dev/null &");
+        ByteArrayOutputStream stdOut = new ByteArrayOutputStream(10);
+        SshMachineLocation machineLocation = (SshMachineLocation) Iterables.getFirst(entity.getLocations(), null);
+        machineLocation.execCommands(
+                ImmutableMap.<String, Object>of(ExecWithLoggingHelpers.STDOUT.getName(), stdOut),
+                "check process is stopped",
+                ImmutableList.of("cat "+driver.getRunDir() + '/' + VanillaSoftwareProcessSshDriver.PID_FILENAME),
```

Comment:
Why are you executing this? Is it really "check process is stopped"? You don't do anything with the response. Do we expect it be running, given that launch has just been called?